aiobotocore/httpchecksum.py

```diff
@@ -155,7 +204,11 @@ async def handle_checksum_body(
 def _handle_streaming_response(http_response, response, algorithm):
     checksum_cls = _CHECKSUM_CLS.get(algorithm)
     header_name = f"x-amz-checksum-{algorithm}"
-    return StreamingChecksumBody(
+    if httpx is not None and isinstance(http_response.raw, httpx.Response):
```

Comment:
Again, this adds some runtime overhead when httpx is installed, but not used. See my other comment.

Reply:
might be able to fix this with #1365 too, with an adapter interface for `httpx.Response`

Reply:
might I suggest for now that we add a common constructor to the http response class since it's ours anyways?